cargo-insta/src/inline.rs

```diff
@@ -4,7 +4,7 @@ use std::fs;
 use std::io::Write;
 use std::path::{Path, PathBuf};
 
-use insta::_cargo_insta_support::SnapshotContents;
+use insta::_cargo_insta_support::TextSnapshotContents;
```

Comment:
I might be missing some context here, but why we need to introduce a separate enum for text? Why not just create a new variant for it? I.e. `SnapshotContents::File`, `SnapshotContents::Inline`, and `SnapshotContents::Binary`?

Reply:
Basically because there is still a lot of shared behavior between file and inline snapshots. By having the `TextSnapshotContents` struct we can use that in the places where we match on the contents and call methods on it. If inline and file were separate variants with separate strings all of this would be much more annoying. You'd have to match with a `|` pattern in the places with shared behavior and then either still call methods on the main `SnapshotContents` and `unwrap` an `Option` it returns or have a freestanding functions to replace the methods that just take a string parameter.

I tried implementing it this way for a bit, but gave up because of all of those problems. It would definitely be doable, but a larger change and likely some additional duplication of code.